waspc/src/Wasp/Message.hs

```diff
@@ -0,0 +1,5 @@
+module Wasp.Message (MessageType (..), SendMessage) where
+
+data MessageType = Start String | Success String | Warning String | Error String
```

Comment:
Based on previous experience, I would name the data constructors here with longer names. That makes it easier to figure out what they are when used in the code (if I see `Start` it can be hard to say what it is, but if I see `StartMessage` it is clearer) and reduces chance of conflict with some other short names. Or, you could leave them as they are, but then I would mostly use them qualified when importing them (`Message.Start`, `Message.Warning`).

Another idea here: we could make these messages more "meaningful". Since compiler has some distinct phases, that are probably not to change so quickly, we could do smth like:
```hs
data CompilerMessage = CompilationStart | CompilationEnd | SetupStart | SetupEnd | Warning String | Error String
```

What this does is give the consumer of these messages more knowledge over what is actually happening. It also delegates construction of (some) actual string messages to the consumer.

If we will be forwarding all messages from the compiler at some point, then we won't really be able to avoid some of them containing `String` as payload, and that is fine, but if we can have some portion of them being more explicit as I described above, that could be a good thing.

What do you think about this? What messages could we have then, does that make sense for the current situation?

Reply:
I've renamed it MessageType, as I had no good reason at all and I don't even do this in other languages. Thanks!

Reply:
I've also postffixed the message types with `Message`. It sounds reasonable, though the invocation points becomes a bit more repetitive with the word `Message` repeated:

```haskell
sendMessage $ SuccessMessage "You succeeded!"
```

I can live with that.

Reply:
Sounds good! Or you could have qualified the import, and you would have `sendMessage $ M.Success "sdfsfd"`. Both are ok with me.